Re-order components of merged file name (#219)
sanger-tol/readmapping: docs/output.md

  - [Short reads](#short-reads) – Aligning HiC and Illumina reads using BWAMEM2 (by default) or MINIMAP2
  - [Oxford Nanopore reads](#oxford-nanopore-reads) – Aligning ONT reads using MINIMAP2
  - [PacBio reads](#pacbio-reads) – Aligning PacBio CLR and CCS filtered reads using MINIMAP2
- [Alignment post-processing](#alignment-post-processing)
  - [Merge by speciemen](#merge-by-speciment) - Merge aligned reads by specimens
  - [External metadata](#external-metadata) – Additional metadata in alignments
  - [Read coverage](#read-coverage) – Read coverage calculations
  - [Statistics](#statistics) – Alignment statistics
- [Workflow reporting](#workflow-reporting)
  - [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution
  - [MultiQC report](#multiqc-report) – Combined input/output QC summary

### Output overview

- `pipeline_info` - execution information of run
- `read_mapping`
- `${datatype}/${specimen}`
  - `${run}/`
    - `${assembly}.${type}.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
    - `${assembly}.${type}.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment

>     - `${assembly}.${type}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format

Reviewer: ```suggestion
    - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
    - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment
    - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format
```

      - `pacbio.${specimen}.${run}.hifitrimmer.summary.json`: HiFi trimmer trimming summary (optional, if filtered reads)
    - `stats/`
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.flagstat`: Number of alignments for each FLAG type
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.idxstats`: Alignment summary statistics
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.stats.gz`: Comprehensive statistics
  - `merged.${#}/` (optional if `params.merged_output` is specified)
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.cram`: Merged aligned CRAM file
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.cram.crai`: Index for the merged alignment
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.coverage.bedGraph.gz`: Read coverage for merged file
    - `stats/`
      - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.flagstat`: Number of alignments for each FLAG type
      - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.idxstats`: Merged alignment summary statistics

>       - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.stats.gz`: Comprehensive statistics for merged alignment

Reviewer: ```suggestion
  - `merged_${#}/` (optional if `params.merged_output` is specified)
    - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.cram`: Merged aligned CRAM file
    - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.cram.crai`: Index for the merged alignment
    - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.coverage.bedGraph.gz`: Read coverage for merged file
    - `stats/`
      - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.flagstat`: Number of alignments for each FLAG type
      - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.idxstats`: Merged alignment summary statistics
      - `${assembly}.${datatype}.${specimen}.merged_${#}.${aligner}.stats.gz`: Comprehensive statistics for merged alignment
```

      - `pacbio.${specimen}.${run}.hifitrimmer.bed.gz`: HiFi trimmer trimming regions (optional, if filtered reads)
      - `pacbio.${specimen}.${run}.hifitrimmer.summary.json`: HiFi trimmer trimming summary (optional, if filtered reads)
    - `stats/`
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.flagstat`: Number of alignments for each FLAG type
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.idxstats`: Alignment summary statistics
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.stats.gz`: Comprehensive statistics
  - `merged.${#}/` (optional if `params.merged_output` is specified)
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.cram`: Merged aligned CRAM file
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.cram.crai`: Index for the merged alignment
    - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.coverage.bedGraph.gz`: Read coverage for merged file
    - `stats/`
      - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.flagstat`: Number of alignments for each FLAG type

>       - `${assembly}.${datatype}.${specimen}.merged.${#}.${aligner}.idxstats`: Merged alignment summary statistics

Reviewer: ```suggestion
```

<details markdown="1">
<summary>Output files</summary>

- `read_mapping/${type}/${specimen}/${run}/qc/`
  - `${type}.${specimen}.${run}.fastqc.html`: An interactive HTML report summarizing key read quality metrics

>   - `${type}.${specimen}.${run}.fastqc.zip`: A compressed archive containing the full set of FASTQC output files

Reviewer: ```suggestion
- `read_mapping/${datatype}/${specimen}/${run}/qc/`
  - `${datatype}.${specimen}.${run}.fastqc.html`: An interactive HTML report summarizing key read quality metrics
  - `${datatype}.${specimen}.${run}.fastqc.zip`: A compressed archive containing the full set of FASTQC output files
```

</details>

### ULI preprocessing

PacBio ULI read (`library`:`uli`) are demultiplexed with LIMA and mark duplicated with PBMARKDUP.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping/pacbio/${specimen}/${run}/qc/`
  - `${datatype}.${specimen}.${run}.pbmarkdup.log`: BED format file with trimming coordinates

>   - `${datatype}.${specimen}.${run}.lima.report`: Statistics of demultiplexing & ULI adpater trimming

Reviewer: ```suggestion
  - `pacbio.${specimen}.${run}.pbmarkdup.log`: BED format file with trimming coordinates
  - `pacbio.${specimen}.${run}.lima.report`: Statistics of demultiplexing & ULI adpater trimming
```

</details>

### Filtering

PacBio reads generated using both CLR and CCS technology are filtered using `HIFITRIMMER`. Additional quality control is performed to check the filtered reads.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping/pacbio/${specimen}/${run}/qc/`
  - `${datatype}.${specimen}.${run}.hifitrimmer.bed.gz`: BED format file with trimming coordinates
  - `${datatype}.${specimen}.${run}.hifitrimmer.summary.json`: Summary statistics of trimming results
  - `${datatype}.${specimen}.${run}.filtered.fastqc.html`: FASTQC report of filtered reads

>   - `${datatype}.${specimen}.${run}.filtered.fastqc.zip`: FASTQC archive of filtered reads

Reviewer: ```suggestion
  - `pacbio.${specimen}.${run}.hifitrimmer.bed.gz`: BED format file with trimming coordinates
  - `pacbio.${specimen}.${run}.hifitrimmer.summary.json`: Summary statistics of trimming results
  - `pacbio.${specimen}.${run}.filtered.fastqc.html`: FASTQC report of filtered reads
  - `pacbio.${specimen}.${run}.filtered.fastqc.zip`: FASTQC archive of filtered reads
```

### Short reads

Short read data from HiC and Illumina technologies is aligned with `BWAMEM2_MEM` (by default) or `MINIMAP2`. The sorted alignment files are processed using the `SAMTOOLS` [mark-duplicate workflow](https://www.htslib.org/algorithms/duplicate.html#workflow). The marked duplicate alignments are output in the CRAM or BAM format.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping`
  - `${datatype}/${specimen}`
    - `${run}/`
      - `${assembly}.${type}.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
      - `${assembly}.${type}.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment

>       - `${assembly}.${type}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format

Reviewer: ```suggestion
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format
```

    - `merged.${#}/` - if params `merge_output`, merged output files with same structure as individual runs, without `qc` folder

</details>

### Oxford Nanopore reads

Reads generated using Oxford Nanopore technology are aligned with `MINIMAP2_ALIGN`. The sorted alignment is output in the CRAM or BAM format.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping`

>   - `ont/${specimen}` - `${run}/` - `${assembly}.ont.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`) - `${assembly}.ont.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment - `${assembly}.ont.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format - `merged.${#}/` - if params `merge_output`.

Reviewer: ```suggestion
  - `ont/${specimen}`
    - `${run}/`
      - `${assembly}.ont.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
      - `${assembly}.ont.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment
      - `${assembly}.ont.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format
    - `merged_${#}/` - if params `merge_output`.
```

  </details>

### PacBio reads

The filtered PacBio reads are aligned with `MINIMAP2_ALIGN`. The sorted alignment is output in the CRAM or BAM format.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping`
  - `pacbio/${specimen}`
    - `${run}/`
      - `${assembly}.pacbio.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
      - `${assembly}.pacbio.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment
      - `${assembly}.pacbio.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format

>     - `merged.${#}/` - if params `merge_output`.

Reviewer: ```suggestion
    - `merged_${#}/` - if params `merge_output`.
```

### Short reads

Short read data from HiC and Illumina technologies is aligned with `BWAMEM2_MEM` (by default) or `MINIMAP2`. The sorted alignment files are processed using the `SAMTOOLS` [mark-duplicate workflow](https://www.htslib.org/algorithms/duplicate.html#workflow). The marked duplicate alignments are output in the CRAM or BAM format.

<details markdown="1">
<summary>Output files</summary>

- `read_mapping`
  - `${datatype}/${specimen}`
    - `${run}/`
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram`: Aligned CRAM file (or `.bam` depending on `--outfmt`)
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.cram.crai`: Index for the alignment
      - `${assembly}.${datatype}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`: Read coverage in bedGraph format

>     - `merged.${#}/` - if params `merge_output`, merged output files with same structure as individual runs, without `qc` folder

Reviewer: ```suggestion
    - `merged_${#}/` - if params `merge_output`, merged output files with same structure as individual runs, without `qc` folder
```

### Read coverage

Read coverage of the output alignment file is calculated with [blobtk depth](https://github.com/genomehubs/blobtk/wiki/blobtk-depth) and output alongside the alignment files.

**File naming:** `${assembly}.${type}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`

### Statistics

The output alignments are used to calculate mapping statistics. Output files are generated using `SAMTOOLS_STATS`, `SAMTOOLS_FLAGSTAT` and `SAMTOOLS_IDXSTATS` and are organized in `stats/` subdirectories of each run or merged specimen:

**File naming:**

- `${assembly}.${type}.${specimen}.${run}.${aligner}.flagstat`: Number of alignments for each FLAG type
- `${assembly}.${type}.${specimen}.${run}.${aligner}.idxstats`: Alignment summary statistics

> - `${assembly}.${type}.${specimen}.${run}.${aligner}.stats.gz`: Comprehensive statistics

Reviewer: ```suggestion
- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.flagstat`: Number of alignments for each FLAG type
- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.idxstats`: Alignment summary statistics
- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.stats.gz`: Comprehensive statistics
```

### Read coverage

Read coverage of the output alignment file is calculated with [blobtk depth](https://github.com/genomehubs/blobtk/wiki/blobtk-depth) and output alongside the alignment files.

**File naming:** `${assembly}.${type}.${specimen}.${run}.${aligner}.coverage.bedGraph.gz`

### Statistics

The output alignments are used to calculate mapping statistics. Output files are generated using `SAMTOOLS_STATS`, `SAMTOOLS_FLAGSTAT` and `SAMTOOLS_IDXSTATS` and are organized in `stats/` subdirectories of each run or merged specimen:

**File naming:**

- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.flagstat`: Number of alignments for each FLAG type
- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.idxstats`: Alignment summary statistics
- `${assembly}.${datatype}.${specimen}.${run}.${aligner}.stats.gz`: Comprehensive statistics

> For merged output (when `merge_output` is enabled), replace `${run}` with `merged.${#}` in the filenames.

Reviewer: ```suggestion
For merged output (when `merge_output` is enabled), replace `${run}` with `merged_${#}` in the filenames.
```
